python/mxnet/gluon/estimator/estimator.py

```diff
@@ -334,6 +355,9 @@ def fit(self, train_data,
             if self.stop_training:
                 break
 
+            # Reset batch size since last batch size may be different
+            batch_size = self.train_stats['batch_size']
```

Comment:
here batch size is more like a hyper parameter, we don't want to change it for the last batch

Reply:
Yes. I have modified the code structure in this commit id [5f94478](https://github.com/apache/incubator-mxnet/pull/14587/commits/5f9447892bfc453963fe6765f171efb556a12649). I am maintaining a separate variable as `current_batch_size`.